### PR TITLE
Add initial component design and review process

### PR DESCRIPTION
**Pull request by rrhyne**
Adds a very brief and introductory description of how a new component should be requested and approved.

### CODE REVIEW

File: handbook/product/design/design_process.md [hunk 1]
```diff
@@ -98,7 +98,12 @@ While asynchronous communication is a core attribute of remote work, key moments
 
 - Visual design should utilize the Sourcegraph’s Figma based component system
 - Dark compositions will be created for each major design
-- If new components are required, components will be created in the file itself, not the component system
+- If new components are required, the following process should be followed
+    - Design component in the Figma file which requires it
+    - @ mention a frontend engineer requesting an addition or change to the component system
+        - Engineering approve, requests changes or request the designer uses an existing component
```
Comment (christinaforney) [inline handbook/product/design/design_process.md:104]:
```suggestion
        - Engineering approves, requests changes, or requests the designer uses an existing component
```

File: handbook/product/design/design_process.md [hunk 2]
```diff
@@ -98,7 +98,12 @@ While asynchronous communication is a core attribute of remote work, key moments
 
 - Visual design should utilize the Sourcegraph’s Figma based component system
 - Dark compositions will be created for each major design
-- If new components are required, components will be created in the file itself, not the component system
+- If new components are required, the following process should be followed
+    - Design component in the Figma file which requires it
+    - @ mention a frontend engineer requesting an addition or change to the component system
+        - Engineering approves, requests changes, or requests the designer uses an existing component
+    - @ mention a designer and request a review
```
Comment (AlicjaSuska) [inline handbook/product/design/design_process.md:105]: We were using Chromatic for this when working with @felixfbecker. I think we should add this detail to the guidelines so it's clear that this is the correct path.

File: handbook/product/design/design_process.md [hunk 3]
```diff
@@ -98,7 +98,12 @@ While asynchronous communication is a core attribute of remote work, key moments
 
 - Visual design should utilize the Sourcegraph’s Figma based component system
 - Dark compositions will be created for each major design
-- If new components are required, components will be created in the file itself, not the component system
+- If new components are required, the following process should be followed
+    - Design component in the Figma file which requires it
```
Comment (AlicjaSuska) [inline handbook/product/design/design_process.md:102]: @rrhyne I would add your requirements that you've mentioned in Figma, here:

1. Proposal with visual examples of Storybook, in product usage and suggested change
2. Sourcegraph search showing affected areas of the product
3. Visual examples of how the proposal affects those areas
4. Github issue for development when the review is complete

**Comment by AlicjaSuska** [comment]
@rrhyne should we merge this?

**Pull request merged**